src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -15,53 +15,70 @@
 #include <ucs/profile/profile.h>
 
 typedef struct {
-    struct mlx5dv_devx_obj     *atomic_dvmr;
+    struct mlx5dv_devx_obj     *dvmr;
     int                        mr_num;
     size_t                     length;
     struct ibv_mr              *mrs[];
 } uct_ib_mlx5_ksm_data_t;
 
+typedef union uct_ib_mlx5_mr {
+    uct_ib_mr_t                super;
+    uct_ib_mlx5_ksm_data_t     *ksm_data;
+} uct_ib_mlx5_mr_t;
```

Comment:
mlx5 data structure refactoring (removing union) should have been a separate PR

Reply:
this PR requires new data structure, code prior to this PR worked with old data structure,
why separate this change?

Reply:
at this point it' too late, but smaller PR's would be faster to review, it could be "preparation" PR just to change data structure